lms/djangoapps/courseware/models.py

```diff
@@ -29,10 +30,41 @@
 log = logging.getLogger("edx.courseware")
 
 
+def chunks(items, chunk_size):
+    """
+    Yields the values from items in chunks of size chunk_size
+    """
+    items = list(items)
+    return (items[i:i + chunk_size] for i in xrange(0, len(items), chunk_size))
+
+
+class ChunkingManager(models.Manager):
+    """
+    :class:`~Manager` that adds an additional method :meth:`chunked_filter` to provide
+    the ability to make select queries with specific chunk sizes.
+    """
+    def chunked_filter(self, chunk_field, items, **kwargs):
+        """
+        Queries model_class with `chunk_field` set to chunks of size `chunk_size`,
+        and all other parameters from `**kwargs`.
```

Comment:
http://stackoverflow.com/questions/6581573/what-are-the-max-number-of-allowable-parameters-per-database-provider-type indicates that there is a limit for each backend (and that sqlite3 is configurable). So, honestly, I'm not sure how best to structure this. I'm going to punt on it for now. Perhaps the real thing that should happen is that the value should be part of the `DATABASES` django settings, with defaults constructed based on the engine type.